### PR TITLE
Trailblazer priority control samples

### PR DESCRIPTION
**Pull request by eliottBo**
## Description
This PR closes [#3889](https://github.com/Clinical-Genomics/cg/issues/3889). The controls currently get express priority in TB. This PR will change this and the control cases will get the priority according to what is stated in statusDB however the slurm QOS will remain equivalent to express.

### Fixed

- Controls ran with slurm qos express have the priority in trailblazer corresponding to the priority in statusDB

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [x] Start a control case and check in TB stage the displayed priority

### Expected test outcome

- [x] Check that the priority is not express in TB
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [x] Tests executed by ELB
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [X] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/workflow/analysis.py [hunk 1]
```diff
@@ -137,6 +137,12 @@ def get_slurm_qos_for_case(self, case_id: str) -> str:
         priority: int = case.priority or Priority.research
         return Priority.priority_to_slurm_qos().get(priority)
 
+    def get_trailblazer_priority(self, case_id: str) -> int:
+        """Get the priority for the case in Trailblazer."""
+        case: Case = self.status_db.get_case_by_internal_id(internal_id=case_id)
+        priority: int = case.priority or Priority.research
```
Comment (islean) [inline cg/meta/workflow/analysis.py:143]: If `case.priority` is an int, then we are basically saying priority = case.priority _unless it is 0, in which instance we should set it to 0_. I think using the case.priority in all cases should work?

File: cg/meta/workflow/utils/utils.py [hunk 2]
```diff
@@ -9,3 +10,14 @@ def are_all_samples_control(case: Case) -> bool:
...
+
+def map_to_trailblazer_priority(priority: int) -> TrailblazerPriority:
+    """Map a priority to a Trailblazer priority."""
+    return {
+        0: TrailblazerPriority.LOW,
+        1: TrailblazerPriority.NORMAL,
+        2: TrailblazerPriority.HIGH,
+        3: TrailblazerPriority.EXPRESS,
+        4: TrailblazerPriority.NORMAL,
+    }[priority]
```
Comment (islean) [inline cg/meta/workflow/utils/utils.py:23]:
```suggestion
MAP_TO_TRAILBLAZER_PRIORITY: dict[int, TrailblazerPriority] = {
        0: TrailblazerPriority.LOW,
        1: TrailblazerPriority.NORMAL,
        2: TrailblazerPriority.HIGH,
        3: TrailblazerPriority.EXPRESS,
        4: TrailblazerPriority.NORMAL,
    }
```
Probably just flavour so feel free to keep it as is. The only difference will be whether you get the priority as `map_to_trailblazer_priority(case.priority)` or `MAP_TO_TRAILBLAZER_PRIORITY[case.priority]`

File: tests/meta/workflow/test_analysis.py [hunk 3]
```diff
@@ -55,6 +55,39 @@ def test_get_slurm_qos_for_case(
...
+        (Priority.research, TrailblazerPriority.LOW),
+        (Priority.standard, TrailblazerPriority.NORMAL),
+        (Priority.priority, TrailblazerPriority.HIGH),
+        (Priority.express, TrailblazerPriority.EXPRESS),
+    ],
+)
+def test_get_trailblazer_priority(
+    case_id: str,
+    priority,
+    expected_trailblazer_priority,
```
Comment (islean) [inline tests/meta/workflow/test_analysis.py:71]:
```suggestion
    priority: Priority,
    expected_trailblazer_priority: TrailblazerPriority,
```

File: cg/meta/workflow/utils/utils.py (hunk 2 above)
Comment (islean) [reply cg/meta/workflow/utils/utils.py:23]: Also - I have no idea how the clinical_trials option 4 should be mapped, but given that it maps to "normal" slurm qos, this probably makes sense

**Review by islean — APPROVED** [review]
Looks good to me!

File: cg/meta/workflow/analysis.py (hunk 1 above)
Comment (eliottBo) [reply cg/meta/workflow/analysis.py]: In the order portal I checked that a case is required to have a priority so I think I can remove it 👍

**Comment by eliottBo** [comment]
# Tests in stage:
## Quality of service express
Screenshot 2024-12-05 at 10 07 45

## Priority High
Screenshot 2024-12-05 at 10 07 13

**Comment by eliottBo** [comment]
# Deployment:
## Stage
Screenshot 2024-12-05 at 10 40 40

Screenshot 2024-12-05 at 10 37 24

## Production:


Screenshot 2024-12-05 at 10 45 56

**Pull request merged**